src/meta/src/hummock/manager/compaction/compaction_group_schedule.rs

```diff
@@ -0,0 +1,316 @@
+        table_ids: &[StateTableId],
+        partition_vnode_count: u32,
+    ) -> Result<CompactionGroupId> {
+        let result = self
+            .move_state_table_to_compaction_group(parent_group_id, table_ids, partition_vnode_count)
+            .await?;
+
+        Ok(result)
+    }
+
+    pub async fn merge_compaction_group(
+        &self,
+        group_1: CompactionGroupId,
+        group_2: CompactionGroupId,
+    ) -> Result<()> {
+        let compaction_guard = self.compaction.write().await;
+        let mut versioning_guard = self.versioning.write().await;
+        let versioning = versioning_guard.deref_mut();
+        // Validate parameters.
+        if !versioning.current_version.levels.contains_key(&group_1) {
+            return Err(Error::CompactionGroup(format!("invalid group {}", group_1)));
+        }
+
+        if !versioning.current_version.levels.contains_key(&group_2) {
+            return Err(Error::CompactionGroup(format!("invalid group {}", group_2)));
+        }
+
+        let state_table_info = versioning.current_version.state_table_info.clone();
+        let member_table_ids_1 = state_table_info
+            .compaction_group_member_table_ids(group_1)
+            .iter()
+            .cloned()
+            .collect_vec();
+
+        let member_table_ids_2 = state_table_info
+            .compaction_group_member_table_ids(group_2)
+            .iter()
+            .cloned()
+            .collect_vec();
+
+        let mut combine_1 = member_table_ids_1.clone();
+        combine_1.extend_from_slice(&member_table_ids_2);
+
+        let mut combine_2 = member_table_ids_2;
+        combine_2.extend_from_slice(&member_table_ids_1);
+
+        if !combine_1.is_sorted() && !combine_2.is_sorted() {
+            return Err(Error::CompactionGroup(format!(
+                "invalid merge group_1 {} group_2 {}",
+                group_1, group_2
+            )));
+        }
+
+        let mut left_group_id = group_1;
+        let mut right_group_id = group_2;
+        let combine_member_table_ids = if combine_1.is_sorted() {
+            combine_1
+        } else {
+            std::mem::swap(&mut left_group_id, &mut right_group_id);
+            combine_2
+        };
+
+        let mut version = HummockVersionTransaction::new(
+            &mut versioning.current_version,
+            &mut versioning.hummock_version_deltas,
+            self.env.notification_manager(),
+            &self.metrics,
+        );
+        let mut new_version_delta = version.new_delta();
+
+        let target_compaction_group_id = {
+            let mut config = self
+                .compaction_group_manager
+                .read()
+                .await
+                .try_get_compaction_group_config(group_1)
+                .unwrap()
+                .compaction_config()
+                .deref()
+                .clone();
+
+            {
+                // update config
+                config.split_weight_by_vnode = 0;
```

Comment:
Please document why we need to set this.

Reply:
typo: remove it